contracts/ThalesRoyale/ThalesRoyale.sol

```diff
@@ -316,50 +369,71 @@ contract ThalesRoyale is Initializable, ProxyOwned, PausableUpgradeable, ProxyRe
     /* ========== INTERNALS ========== */
 
     function _signUpPlayer(address _player, uint _position) internal {
+        uint tokenId = thalesRoyalePassport.safeMint(_player);
+        tokenSeason[tokenId] = season;
+
+        tokensMintedPerSeason[season][tokenId] = block.timestamp;
+        tokensPerSeason[season].push(tokenId);
+        mintedTokensCount[season]++;
+
         playerSignedUpPerSeason[season][_player] = block.timestamp;
-        playersPerSeason[season].push(_player);
-        signedUpPlayersCount[season]++;
 
         if (_position != 0) {
             for(uint i = 1; i <= rounds; i++){
-                _putPosition(_player, season, i, _position);
+                _putPosition(_player, season, i, _position, tokenId);
             }
         }
 
         _buyIn(_player, buyInAmount);
 
-        emit SignedUp(_player, season, _position);
+        emit SignedUp(_player, tokenId, season, _position);
```

Comment:
i would create new message because this one is extended, we had issue for extending messages on the subgraph and building it because it is indexed. So, better to create new one `SignedUpPassport` or however you want to call it 😆